sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -240,10 +240,52 @@ class Analyzer(
           x.child
         }
 
+        // We will insert another Projection if the GROUP BY keys are contained in the
```

Comment:
Maybe in addition to describing what we do we should explain why we are inserting this projection here as well?